test/unit/Trader.js

```diff
@@ -0,0 +1,99 @@
+const { expect } = require("chai")
+const { ethers, getNamedAccounts, deployments } = require("hardhat")
+const { deploy } = deployments
+const { signOrders } = require("@tracer-protocol/tracer-utils")
+
+describe("Unit tests: Trader.sol", function () {
+    let trader
+    let accounts
+    let perpMockAddress
+
+    before(async function () {
+        const { deployer } = await getNamedAccounts()
+
+        libPerpetuals = await deploy("Perpetuals", {
+            from: deployer,
+            log: true,
+        })
+
+        await deploy("Trader", {
+            from: deployer,
+            log: true,
+            libraries: {
+                Perpetuals: libPerpetuals.address,
+            },
+        })
+
+        await deploy("TracerPerpetualSwapMock", {
+            from: deployer,
+            log: true,
+        })
+
+        perpMockAddress = await deployments.get("TracerPerpetualSwapMock")
+
+        let traderDeployment = await deployments.get("Trader")
+        trader = await ethers.getContractAt(
+            traderDeployment.abi,
+            traderDeployment.address
+        )
+        accounts = await ethers.getSigners()
+    })
+
+    describe("executeTrader", async () => {
+        context("When the makers array is empty", async () => {
+            it("reverts", async () => {
+                await expect(trader.executeTrade([], [])).to.be.revertedWith(
+                    "TDR: Received empty arrays"
+                )
+            })
+        })
+        context("When the takers array is empty", async () => {
+            it("reverts", async () => {
+                await expect(trader.executeTrade([], [])).to.be.revertedWith(
+                    "TDR: Received empty arrays"
+                )
+            })
+        })
+        context("When the maker and taker array lengths differ", async () => {
+            it("reverts", async () => {
+                await expect(trader.executeTrade([], [])).to.be.revertedWith(
+                    "TDR: Received empty arrays"
+                )
+            })
+        })
+        context("When a single order signature is incorrect", async () => {
+            it("skips that order pairing", async () => {})
+        })
+        context("When an order already exists", async () => {
+            it("does not create a new order", async () => {})
+
+            it("processes the order as is", async () => {})
```

Comment:
Why do we want it to "process the order as is" if an identical one already exists?

Or will the second test here make the same order with a different `created` time?

Reply:
I guess the wording isn't 100% clear. What I want to test here is if the contract has already received an order before, it doesn't "recreate" that order. It will simply check how much of it has been filled and execute it against its paired order if possible.